### PR TITLE
Story CORE-289 | CLI authentication

### PR DESCRIPTION
**Pull request by franciscobonand**
# Description

Kind: Auth / CLI

Section: CLI

Summary: Inserts authentication in CLI commands, as well as authentication errors handling

Issue link: https://inspr.atlassian.net/browse/CORE-289

## Changelog

- Many things

### CODE REVIEW

File: pkg/cmd/utils/cli_client.go [hunk 1]
```diff
@@ -50,10 +52,31 @@ func SetOutput(out io.Writer) {
...
+
+// RequestErrorMessage prints an error to the user based on the error given, in
+// actuality it converts the error to an insprErr and then process what type
+// of return the apply request returned.
+func RequestErrorMessage(err error, w io.Writer) {
+	ierr, ok := err.(*ierrors.InsprError)
+	if ok {
+		switch ierr.Code {
+		case ierrors.Unauthorized:
+			fmt.Fprintf(w, "did you login ?\n")
```
Comment (brenordr) [inline pkg/cmd/utils/cli_client.go:73]: Maybe use a more descriptive message?
This seems too dry.

**Review by brenordr — APPROVED** [review]
LGTM

File: cmd/inspr/cli/auth_init.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	"context"
+	"fmt"
+
+	"github.com/inspr/inspr/pkg/cmd"
+	"github.com/inspr/inspr/pkg/cmd/utils"
+)
+
+var authCommand = cmd.NewCmd("cluster").AddSubCommand(
+	cmd.NewCmd("init").
+		WithCommonFlags().ExactArgs(1,
```
Comment (franciscobonand) [inline cmd/inspr/cli/auth_init.go]: I'm not sure but I think the only flag configured as a Common Flag is "scope", and I don't think it applies here

File: cmd/insprd/memory/tree/dapp.go [hunk 3]
```diff
@@ -210,7 +210,7 @@ func (amm *AppRootGetter) Get(query string) (*meta.App, error) {
 			nextApp = nextApp.Spec.Apps[element]
 			if nextApp == nil {
 				logger.Error("unable to find dApp for given query (Root Getter)",
-					zap.String("query", query))
+					zap.String("query", string(query)))
```
Comment (franciscobonand) [inline cmd/insprd/memory/tree/dapp.go]: Isn't query already a string?

**Review by LucasStarlingdePaulaSalles — APPROVED** [review]

**Review by nicholaspcr — APPROVED** [review]

File: build/uidp_helm/values.yaml [hunk 4]
```diff
@@ -41,6 +41,7 @@ secretRepository: gcr.io/red-inspr/uidp/redis/secret
 secret:
   privateKeyName: redisprivatekey
   adminPassword: "123456"
+  adminToken: eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2MTkxOTI5ODYsInBheWxvYWQiOnsidWlkIjoiIiwicm9sZSI6MSwic2NvcGUiOlsiIl0sInJlZnJlc2giOm51bGwsInJlZnJlc2h1cmwiOiIifX0.V08xa5SDLTEOsSJnQlpVGZOyBHq8fKGAGjLY1sQwc9FK-u4pePJSbYXsCAPJQOai-LR55NYcAOk-dsMTS6RRYQ # set this admin token from a token that contains full power over the cluster
```
Comment (brenordr) [inline build/uidp_helm/values.yaml:44]: @ptcar2009 Document the flow of uuid provider 😄

File: cmd/insprd/operators/kafka/nodes/converter.go [hunk 5]
```diff
@@ -159,6 +159,8 @@ func (no *NodeOperator) dAppToDeployment(app *meta.App) *kubeApp.Deployment {
 	}
 }
 
+const sidecarPort int32 = 8888
```
Comment (brenordr) [inline cmd/insprd/operators/kafka/nodes/converter.go]: 3051 or var in the helm

File: pkg/auth/jwt/auth.go [hunk 6]
```diff
@@ -42,34 +43,31 @@ func NewJWTauth(rsaPublicKey *rsa.PublicKey) *JWTauth {
...
+		if err.Error() == errors.New(`exp not satisfied`).Error() {
+
+			newToken, err := JA.Refresh(token)
+			if err != nil {
+				return nil,
+					token,
+					ierrors.
+						NewError().
+						InternalServer().
+						Message("error refreshing token").
```
Comment (brenordr) [inline pkg/auth/jwt/auth.go:62]: @ptcar2009 Add more description why the token failed to refresh

**Review by brenordr — APPROVED** [review]
LGTM

**Pull request merged**